### PR TITLE
Scheduled Transactions: MerkleSchedule

### PR DESCRIPTION
**Pull request by failfmi**
**Related issue(s)**:
Closes #None

**Summary of the change**:
* Add MerkleSchedule
* State: Add FCMap<MerkleEntityId, MerkleSchedule>

**External impacts**:
None.

**Applicable documentation**
- [ ] Javadoc
- [ ] HAPI protobuf comments
- [ ] README

### CODE REVIEW

**Review by tinker-michaelj — CHANGES_REQUESTED** [review]
Looks great overall, made some comments to avoid future work needed to remove obsolete code elements. 👍

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleSchedule.java [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+            Map<EntityId, Boolean> signers,
+            Map<EntityId, byte[]> signatures
+    ) {
+        this.transactionBody = transactionBody;
+        this.signersThreshold = signersThreshold;
+        this.signers = signers;
+        this.signatures = signatures;
+    }
+
+    @Deprecated
```
[the remark this answers is not in the document]
Comment (failfmi) [reply hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleSchedule.java:82]: Will resolve the issue in a separate PR.

File: hedera-node/src/main/java/com/hedera/services/ServicesState.java [hunk 2]
```diff
@@ -160,6 +165,11 @@ public void initialize(MerkleInternal previous) {
 			log.info("Created disk file system after <=0.9.0 state restoration");
 			skipDiskFsHashCheck = true;
 		}
+		if (scheduleTxs() == null) {
+			setChild(ChildIndices.SCHEDULE_TXS,
+					new FCMap<>(new MerkleEntityId.Provider(), MerkleSchedule.LEGACY_PROVIDER));
+			log.info("Created scheduled txs FCMap after <= 1.0.0 state restoration");
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/ServicesState.java:171]: In a future PR could fix typo `<= 1.0.0` as `<=0.10.0`. 👍

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Comment by tinker-michaelj** [comment]
After running CodeCov, we need to add unit tests before that check will pass - see above.

**Review by tinker-michaelj — APPROVED** [review]
LGTM...seems as if CodeCov isn't picking up all the test coverage after running CI, will see what happens next.

**Pull request merged**